src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -75,7 +75,11 @@ protected Workspace(HostServices host, string? workspaceKind)
             _services = host.CreateWorkspaceServices(this);
 
             _legacyOptions = _services.GetRequiredService<ILegacyWorkspaceOptionService>();
-            _legacyOptions.RegisterWorkspace(this);
+
+            if (CanUpdateOptions)
+            {
+                _legacyOptions.RegisterWorkspace(this);
+            }
```

Comment:
This might need a comment as to what's going on here -- like I get it knowing how it all works, but anybody else might wonder why the check is here.